static/js/src/app/settings.js

```diff
@@ -21,9 +22,228 @@ $(document).ready(function () {
+                $("#folder").attr("disabled", true);
+                $("#restrictdomain").attr("disabled", true);
+                $('#deletecampaign').attr("disabled", true);
+                $('#lastlogin').attr("disabled", true);
+                $('#imapfreq').attr("disabled", true);
+        $("#validateimap").attr("disabled", true);  
+        $("#validateimap").html("<i class='fa fa-circle-o-notch fa-spin'></i> Testing...");
+        
+        api.IMAP.validate(server).done(function(data) {
+            if (data.success == true) {
+                Swal.fire({
+                    title: "Success",
+                    html: "Logged into <b>" + $("#imaphost").val() + "</b>",
+                    type: "success",
+                })
+            } else {
+                Swal.fire({
+                    title: "Failed!",
+                    html: "Unable to login to <b>" + $("#imaphost").val() + "</b>.",
+                    type: "error",
+                    showCancelButton: true,
+                    cancelButtonText: "Close",
+                    confirmButtonText: "More Info",
+                    confirmButtonColor: "#428bca",
+                    allowOutsideClick: false,
+                }).then((result) => {
+                    Swal.fire({
+                        title: "Error:",
+                        text: data.message,
+                      })
+                  })
+            }
+            
+          })
+          .fail(function() {
+            Swal.fire({
+                title: "Failed!",
+                text: "An unecpected error occured.",
+                type: "error",
+            })
+          })
+          .always(function() {
+            //Re-enable inputs and change button text
+            $("#imaphost").attr("disabled", false);
+            $("#imapport").attr("disabled", false);
+            $("#imapusername").attr("disabled", false);
+            $("#imappassword").attr("disabled", false);
+            $("#use_imap").attr("disabled", false);
+            $("#use_tls").attr("disabled", false);
+            $("#folder").attr("disabled", false);
+            $("#restrictdomain").attr("disabled", false);
+            $('#deletecampaign').attr("disabled", false);
+            $('#lastlogin').attr("disabled", false);
+            $('#imapfreq').attr("disabled", false);
+            $("#validateimap").attr("disabled", false);
+            $("#validateimap").html(oldHTML);
+
+          });
+
+      }); //end testclick
+
+    $("#reporttab").click(function() {
+        loadIMAPSettings()
+    })
+
+    $("#advanced").click(function() {
+        $("#advancedarea").toggle();
+    })
+
+    function lastLoginTicker(){
+
+        if ($("#lastlogin").is(":visible") && $('#use_imap').prop('checked') == true) {
+            api.IMAP.get()
+            .success(function (imap) {
+                if (imap.length > 0){
+                    $('#lastlogin').val(imap[0].last_login_friendly)
```

Comment:
```suggestion
                    $('#lastlogin').val(moment.utc(imap[0].last_login).fromNow())
```

Reply:
Done. I removed the ticker altogether. I also created a hidden field in the HTML page to store the 'raw' last_login value in order to pass it back to the server in the save operation. In hindsight I maybe should have used Gorm's Update, as I think that only updates changed fields.

Also fixed a bug where one successful login updated all users' successful logins.

Commit: https://github.com/gophish/gophish/pull/1612/commits/2cdc12e94e3ad7473ffc84a0d3a7098521439ab4